### PR TITLE
Feat 2.5: Adding and documenting endpoints

### PR DESCRIPTION
**Pull request by jamkerr**
The endpoints added and documented are to:
- return all movies
- return one movie
- return a genre
- return a director
- add a new user
- update a username
- add a favorite movie
- remove a favorite movie
- delete a user

Known bug: removing a favorite movie doesn't work properly yet

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -45,18 +49,134 @@ let topTenMovies = [
...
+// Remove a favorite movie
+app.delete('/users/:username/favorites', (req, res) => {
+    // Find user by username
+    let user = users.find((user) => { return user.username === req.params.username });
+    // Favorite to delete
+    let notFavoriteAnymore = req.body.title;
+
+    // If user exists and they have the film as a favorite
+    if (user) {
+        let isFavorite = user.favorites.find((favorite) => { return favorite.title === notFavoriteAnymore.title });
```
Comment (oxenprogrammer) [inline index.js]: `notFavoriteAnymore` on line 149 is already `req.body.title`. The bug here is that we need to remove the `.title`

File: index.js [hunk 2]
```diff
@@ -45,18 +49,134 @@ let topTenMovies = [
...
+        if (isFavorite) {
+            user.favorites.delete(notFavoriteAnymore);
```
Comment (oxenprogrammer) [inline index.js]: We do not have any JavaScript method `Object.delete` we rather have `delete object`. In this case we will need to probably create a helper function and then inside it iterate through the object and when we find the key with the value `notFavoriteAnymore` we delete it.

**Review by oxenprogrammer — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (jamkerr) [reply index.js]: Thanks for picking that up!

File: index.js (hunk 2 above)
Comment (jamkerr) [reply index.js]: Ahh that makes sense. I went with the same filtering solution that I use to delete users.

File: index.js [hunk 3]
```diff
@@ -150,12 +151,12 @@ app.delete('/users/:username/favorites', (req, res) => {
 
     // If user exists and they have the film as a favorite
     if (user) {
-        let isFavorite = user.favorites.find((favorite) => { return favorite.title === notFavoriteAnymore.title });
+        let isFavorite = user.favorites.find((favorite) => { return favorite.title === notFavoriteAnymore });
         if (isFavorite) {
-            user.favorites.delete(notFavoriteAnymore);
-            res.status(201).send(`You removed ${notFavoriteAnymore.title} as a favorite.`);
+            user.favorites = user.favorites.filter((obj) => { return obj.title !== notFavoriteAnymore });
+            res.status(201).send(`You removed ${notFavoriteAnymore} as a favorite.`);
```
Comment (oxenprogrammer) [inline index.js]: The status code for deleted is `204` that actually returns no content. `201` is for created successfully. Please take a look [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status) on when to use a specific status code.

**Review by oxenprogrammer — COMMENTED** [review]
`LGTM` except the status code.

**Pull request merged**